SpaceGame/prototype.py

```diff
@@ -0,0 +1,205 @@
+      for self.__y in range(0,getHeight(newShip)):
+        oldX = self.__x * 2
+        oldY = self.__y * 2
+        newPixel = getPixelAt(newShip, self.__x, self.__y)
+        oldPixel = getPixelAt(self.__ship, oldX, oldY)
+        setColor(newPixel, getColor(oldPixel))
+    self.__ship = newShip
+    return self.__ship
+    
+  def paintShip(self):                                     #this definition works out where to use the small image green screen script and runs it
+    xPos = getWidth(self.__space)/2 - getWidth(self.__ship)/2
+    yPos = getHeight(self.__space)/2 - getHeight(self.__ship)/2
+    return GreenScreen(self.__ship, self.__replaceColour, self.__threshold, self.__space).screenSmallImage(xPos, yPos)
+ 
+  def gameOver(self):                                      #this is the final game over animation                                     
+    numBars = getWidth(self.__space)/7
+    for index in range(0, 5):
+      dyingImage = self.paintShip()
+      self.shrinkShip()
+      finalImage = GlitchEffect(dyingImage, numBars).run()
+      numBars = numBars*2
+      repaint(finalImage)
+      copyPicture(self.__spaceBackup, self.__space)        #had to use this script as the duplicate picture feature breaks the repaint feature
+    GreenScreen(makePicture("""SpaceGame\gameOverText.jpg"""), black, 100, self.__space).sameImageSize()
+    repaint(self.__space)
+    
+def main():
+  #in ship locations
+  inShipSpace = GreenScreen(makePicture("""SpaceGame\cockPit.jpg"""), green, 170, makePicture("""SpaceGame\space.jpg""")).sameImageSize()
+  inShipMoon = GreenScreen(makePicture("""SpaceGame\cockPit.jpg"""), green, 170, makePicture("""SpaceGame\moon.jpg""")).sameImageSize()
+  inShipPlanet = GreenScreen(makePicture("""SpaceGame\cockPit.jpg"""), green, 170, makePicture("""SpaceGame\planet.jpg""")).sameImageSize()
+  
+  #game over setup
+  inSpace = LostInSpace(makePicture("""SpaceGame\spaceShip.jpg"""), makePicture("""SpaceGame\space.jpg"""), red, 150)
+  
+  #left ship scenario's
+  onPlanet = Scroll(makePicture("""SpaceGame\planet.jpg"""), 10)
+  onMoon = Scroll(makePicture("""SpaceGame\moon.jpg"""), 10)
+  
+  #first screen
+  GreenScreen(makePicture("""SpaceGame\cockPit.jpg"""), green, 170, makePicture("""SpaceGame\space.jpg""")).animate()
+  imageToRender = makeEmptyPicture(250,250)
+  #game loop
+  while true:
+    command = requestIntegerInRange("Press \"One\" to go to a planet \nPress \"Two\" to go to a moon \nPress \"Three\" to leave your ship", 1, 3)
+    if command == 1:
+      copyPicture(inShipPlanet, imageToRender)            #this stage was before every the repaint function to try and make sure a new window isn't opened everytime
+      repaint(imageToRender)
+      command = requestIntegerInRange("Press \"One\" to go to space \nPress \"Two\" to leave your ship", 1, 2)
+      if command == 2:
+        onPlanet.animate(10)
+        command = requestString("Press any key to go to space")
+        command = 0
+    if command == 2:
```

Comment:
Looking back at this -- it seems to be okay. Not sure why it renders incorrectly on the other screen.